uniffi/src/interface/mod.rs

```diff
@@ -8,36 +8,36 @@
 
 //! # Component Interface Definition and Types
```

Comment:
The changes in this file are broadly:

* Modifying the ffi-level functions to use ffi-level types.
* Adding the new string-alloc helper function.
* Adding some docs and general code cleanup while I was in here.